pg_shard--1.1--1.2.sql

```diff
@@ -1,13 +1,242 @@
+				   shardminvalue AS min_value,
+				   shardmaxvalue AS max_value
+			FROM   pg_dist_shard;
+
+		CREATE TRIGGER shard_insert
+		INSTEAD OF INSERT ON pgs_distribution_metadata.shard
+			FOR EACH ROW
+			EXECUTE PROCEDURE adapt_and_insert_shard();
+
+		CREATE VIEW pgs_distribution_metadata.shard_placement AS
+			SELECT oid::bigint AS id,
+				   shardid     AS shard_id,
+				   shardstate  AS shard_state,
+				   nodename    AS node_name,
+				   nodeport    AS node_port
+			FROM   pg_dist_shard_placement;
+
+		CREATE TRIGGER shard_placement_insert
+		INSTEAD OF INSERT ON pgs_distribution_metadata.shard_placement
+			FOR EACH ROW
+			EXECUTE PROCEDURE adapt_and_insert_shard_placement();
+
+		CREATE VIEW pgs_distribution_metadata.partition AS
+			SELECT logicalrelid AS relation_id,
+				   partmethod   AS partition_method,
+				   column_to_column_name(logicalrelid, partkey) AS key
+			FROM   pg_dist_partition;
+
+		CREATE TRIGGER partition_insert
+		INSTEAD OF INSERT ON pgs_distribution_metadata.partition
+			FOR EACH ROW
+			EXECUTE PROCEDURE adapt_and_insert_partition();
+	ELSE
+		-- add default values to id columns
+		ALTER TABLE pgs_distribution_metadata.shard
+			ALTER COLUMN id
+			SET DEFAULT nextval('pgs_distribution_metadata.shard_id_sequence');
+		ALTER TABLE pgs_distribution_metadata.shard_placement
+			ALTER COLUMN id
+			SET DEFAULT nextval('pgs_distribution_metadata.shard_placement_id_sequence');
+
+		-- associate sequences with their columns
+		ALTER SEQUENCE pgs_distribution_metadata.shard_id_sequence
+			OWNED BY pgs_distribution_metadata.shard.id;
+		ALTER SEQUENCE pgs_distribution_metadata.shard_placement_id_sequence
+			OWNED BY pgs_distribution_metadata.shard_placement.id;
+	END IF;
+END;
+$$;
+
+-- Syncs rows from the pg_shard distribution metadata related to the specified
+-- table name into the metadata tables used by CitusDB. After a call to this
+-- function for a particular pg_shard table, that table will become usable for
+-- queries within CitusDB. If placement health has changed for given pg_shard
+-- table, calling this function an additional time will propagate those health
+-- changes to the CitusDB metadata tables.
+CREATE OR REPLACE FUNCTION sync_table_metadata_to_citus(table_name text)
+RETURNS void
+AS $sync_table_metadata_to_citus$
+	DECLARE
+		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
+		dummy_shard_length CONSTANT bigint := 0;
+		warning_msg CONSTANT text := 'sync_table_metadata_to_citus is deprecated and ' ||
+									 'will be removed in a future version';
+	BEGIN
+		RAISE WARNING '%', warning_msg;
+
+		-- grab lock to ensure single writer for upsert
+		LOCK TABLE pg_dist_shard_placement IN EXCLUSIVE MODE;
+
+		-- First, update the health of shard placement rows already copied
+		-- from pg_shard to CitusDB. Health is the only mutable attribute,
+		-- so it is presently the only one needing the UPDATE treatment.
+		UPDATE pg_dist_shard_placement
```

Comment:
This function already exists in `pg_shard` and is not changing in this release. The only change here is to add the `RAISE WARNING` as users do not need to call this method any longer now that metadata just stays in sync.

In v1.3 we'll remove this function entirely, but we need it in v1.2 to aid in migration from 1.1 to 1.2, since CitusDB + `pg_shard` users need this workflow to upgrade:

  * Perform one last sync from `pg_shard` to CitusDB
  * Drop `pg_shard` metadata tables
  * Recreate `pg_shard` metadata relations as views into CitusDB
  * Add warning to sync method to let users know they no longer need it

_Technically_ we could remove this function altogether right now as the last step, but if we're being strict about version numbers that's a breaking change and would necessitate a bump to v2.0. So we'll have a deprecation warning until then.